src/Transport/Receiving/MessagePump.cs

```diff
@@ -35,63 +31,104 @@ class MessagePump : IPushMessages
         int maxConcurrency;
         MessageReceiver receiver;
 
-        static readonly ILog logger = LogManager.GetLogger<MessagePump>();
+        static readonly ILog Logger = LogManager.GetLogger<MessagePump>();
 
-        public MessagePump(ServiceBusConnectionStringBuilder connectionStringBuilder, ITokenProvider tokenProvider, int prefetchMultiplier, int? overriddenPrefetchCount,
-            TimeSpan timeToWaitBeforeTriggeringCircuitBreaker, RetryPolicy retryPolicy)
+        PushRuntimeSettings limitations;
+
+        public MessagePump(
+            ServiceBusConnectionStringBuilder connectionStringBuilder,
+            AzureServiceBusTransport transportSettings,
+            ReceiveSettings receiveSettings,
+            Action<string, Exception> criticalErrorAction,
+            NamespacePermissions namespacePermissions,
+            QueueCreator queueCreator)
         {
+            Id = receiveSettings.Id;
             this.connectionStringBuilder = connectionStringBuilder;
-            this.tokenProvider = tokenProvider;
-            this.prefetchMultiplier = prefetchMultiplier;
-            this.overriddenPrefetchCount = overriddenPrefetchCount;
-            this.timeToWaitBeforeTriggeringCircuitBreaker = timeToWaitBeforeTriggeringCircuitBreaker;
-            this.retryPolicy = retryPolicy;
+            this.transportSettings = transportSettings;
+            this.receiveSettings = receiveSettings;
+            this.criticalErrorAction = criticalErrorAction;
+            this.queueCreator = queueCreator;
+
+            if (receiveSettings.UsePublishSubscribe)
+            {
+                Subscriptions = new SubscriptionManager(
+                    receiveSettings.ReceiveAddress,
+                    transportSettings,
+                    connectionStringBuilder,
+                    namespacePermissions);
+            }
         }
 
-        public Task Init(Func<MessageContext, Task> onMessage, Func<ErrorContext, Task<ErrorHandleResult>> onError, CriticalError criticalError, PushSettings settings)
+        public async Task Initialize(
+            PushRuntimeSettings limitations,
+            Func<MessageContext, Task> onMessage,
+            Func<ErrorContext, Task<ErrorHandleResult>> onError)
         {
-            if (settings.PurgeOnStartup)
+            if (receiveSettings.PurgeOnStartup)
             {
                 throw new Exception("Azure Service Bus transport doesn't support PurgeOnStartup behavior");
             }
 
+            if (Subscriptions != null)
+            {
+                await queueCreator.CreateSubscription(receiveSettings.ReceiveAddress).ConfigureAwait(false);
```

Comment:
should the `CreateSubscription` API be moved to the subscription manager instead?